sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -31,64 +31,340 @@
 
 import operator
 import pandas
+import numpy
+
+import sdc
 
 from numba import types
 from numba.extending import (overload, overload_method, overload_attribute)
+from sdc.hiframes.pd_dataframe_ext import DataFrameType
 from numba.errors import TypingError
 
-from sdc.datatypes.hpat_pandas_dataframe_types import DataFrameType
+if not sdc.config.use_default_dataframe:
+    from sdc.datatypes.hpat_pandas_dataframe_types import DataFrameType
+
+
+    @overload_method(DataFrameType, 'count')
+    def sdc_pandas_dataframe_count(self, axis=0, level=None, numeric_only=False):
+        """
+        Pandas DataFrame method :meth:`pandas.DataFrame.count` implementation.
+
+        .. only:: developer
+
+            Test: python -m sdc.runtests sdc.tests.test_dataframe.TestDataFrame.test_count
+
+        Parameters
+        -----------
+        self: :class:`pandas.DataFrame`
+            input arg
+        axis:
+            *unsupported*
+        level:
+            *unsupported*
+        numeric_only:
+            *unsupported*
+
+        Returns
+        -------
+        :obj:`pandas.Series` or `pandas.DataFrame`
+                returns: For each column/row the number of non-NA/null entries. If level is specified returns a DataFrame.
+        """
+
+        _func_name = 'Method pandas.dataframe.count().'
+
+        if not isinstance(self, DataFrameType):
+            raise TypingError('{} The object must be a pandas.dataframe. Given: {}'.format(_func_name, self))
+
+        if not (isinstance(axis, types.Omitted) or axis == 0):
+            raise TypingError("{} 'axis' unsupported. Given: {}".format(_func_name, axis))
+
+        if not (isinstance(level, types.Omitted) or level is None):
+            raise TypingError("{} 'level' unsupported. Given: {}".format(_func_name, axis))
+
+        if not (isinstance(numeric_only, types.Omitted) or numeric_only is False):
+            raise TypingError("{} 'numeric_only' unsupported. Given: {}".format(_func_name, axis))
+
+        def sdc_pandas_dataframe_count_impl(self, axis=0, level=None, numeric_only=False):
+            result_data = []
+            result_index = []
+
+            for dataframe_item in self._data:
+                item_count = dataframe_item.count()
+                item_name = dataframe_item._name
+                result_data.append(item_count)
+                result_index.append(item_name)
+
+            return pandas.Series(data=result_data, index=result_index)
+
+        return sdc_pandas_dataframe_count_impl
+
+else:
+    def reduce(df, name):
+        saved_columns = df.columns
+        n_cols = len(saved_columns)
+        data_args = tuple('data{}'.format(i) for i in range(n_cols))
+        func_text = "def _reduce_impl(df, axis=None, skipna=None, level=None, numeric_only=None):\n"
```

Comment:
This line defines function signature, which is now actually hardcoded to signature that 'median' has (the other function like max, min, mean also have the same signature, but not all). That won't work for other functions, e.g. Series.sum, which has additional arguments:
   Series.sum(self, axis=None, skipna=None, level=None, numeric_only=None, **min_count=0,** **kwargs)
You have to pass additional arguments to this function to be able to generate correct function text that matches the signature used in overload.